app/controllers/roles_controller.rb

```diff
@@ -36,14 +35,17 @@ def create
     if @role.save
       process_success
     else
+      if cloning?
+        @cloned_role = true
+        @original_role_id = params[:original_role_id]
+      end
       process_error
     end
   end
 
   def clone
     @cloned_role      = true
     @original_role_id = @role.id
-    notice(_("Role cloned from role %{old_name}") % { :old_name => @role.name })
```

Comment:
this action is the same as "new" action, it displays the form and only prepopulates it with some original role data, at this time it should not display any notification since no object was saved yet (I know it's unrelated but I was hoping that small fixes like this could squeeze in)